### PR TITLE
Feature: Refine documentation and optimize the usage of configs

### PR DESCRIPTION
**Pull request by wwwy3y3**
## Description
Refine documentation and optimize the usage of configs.

### CODE REVIEW

File: packages/doc/docs/api-plugin/rate-limit.mdx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+| Setting  | Default | Type    | Description                                                 |
+| -------- | ------- | ------- | ----------------------------------------------------------- |
+| enabled  | true    | boolean | Enable or disable rate limit service (true to enable).     |
+
+## Options
+
+| Option   | Default | Type  | Description                                                            |
+| -------- | ------- | ----- | ---------------------------------------------------------------------- |
+| interval | null    | time  | Length of time request records are kept in memory (default: 1 minute). |
+| max      | 60      | number| Maximum number of connections during `interval` before sending a 429.  |
```
Comment (onlyjackfrost) [inline packages/doc/docs/api-plugin/rate-limit.mdx:25]: I think `before sending a HTTP 429` would be better

**Review by onlyjackfrost — COMMENTED** [review]
Besides the typo, Other LGTM

File: packages/doc/docs/data-privacy/cls.mdx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Column Level Security
+Column Level Security (CLS) is a data security measure that provides fine-grained control over access to specific columns of a database table. With CLS, database administrators can restrict access to sensitive data by allowing only authorized users or roles to access specific columns. This helps to protect against data leaks or unauthorized access to sensitive information.
+
+In the VulcanSQL system, CLS can be achieved by utilizing if-else conditions in conjunction with user attributes, which are accessible through context.user.attr in VulcanSQL. For instance, to limit access to the salary column in the employees table, you may use the following query:
```
Comment (fredalai) [inline packages/doc/docs/data-privacy/cls.mdx:4]:
```suggestion
In the VulcanSQL system, CLS can be achieved by utilizing if-else conditions in conjunction with user attributes, which are accessible through `{{ context.user.attr }}` in VulcanSQL. For instance, to limit access to the salary column in the employees table, you may use the following query:
```

File: packages/doc/docs/api-plugin/pagination.mdx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+## Implement Manually
+
+You can implement your own pagination strategies by modifying your SQL queries. For example, you can use parameters to implement a simple offset-based pagination:
+
+```sql
+SELECT * FROM users
+-- highlight-next-line
+LIMIT {{ context.params.limit }}
+-- highlight-next-line
+OFFSET {{ context.params.offset }}
```
Comment (fredalai) [inline packages/doc/docs/api-plugin/pagination.mdx:15]:
```suggestion
SELECT * FROM users
// highlight-next-line
LIMIT {{ context.params.limit }}
// highlight-next-line
OFFSET {{ context.params.offset }}
```

File: packages/doc/docs/get-started/first-api.mdx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+  ```bash
+  vulcan init --template=quick-start-from-binary
+  ```
+
+  </TabItem>
+  <TabItem value="nodejs" label="NodeJS Project">
+
+  ```bash
+  vulcan init --template=quick-start-from-node
```
[the remark this answers is not in the document]
Comment (wwwy3y3) [reply packages/doc/docs/get-started/first-api.mdx:25]: I'm doing it.

File: packages/doc/docs/get-started/installation.mdx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## Install From Binary
+We provide pre-built binaries for Linux, macOS, and Windows. The binary includes all dependencies, so you don't need to install any additional dependencies.
+
+You can find all versions on [GitHub Releases](https://github.com/Canner/vulcan-sql/releases)
+
+<Tabs groupId="install-os">
+<TabItem value="osx" label="MacOS">
+
+Download the latest version of VulcanSQL CLI from: [vulcan](./).
```
[the remark this answers is not in the document]
Comment (wwwy3y3) [reply packages/doc/docs/get-started/installation.mdx:16]: yeah, planned to put it in once I confirm how the link look like in GitHub artifact

File: packages/doc/docs/get-started/first-api.mdx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  </TabItem>
+  </Tabs>
+
+  Once the project is initialized, start the server with the following command:
+
+  ```bash
+  vulcan start --watch
+  ```
+
+  In this quick-start project, we query the flights data from a CSV file with DuckDB. To test the API endpoint, try the following query in your browser or with curl:
```
[the remark this answers is not in the document]
Comment (wwwy3y3) [reply packages/doc/docs/get-started/first-api.mdx:37]: perhaps I'll just use `in this tutorial`

File: packages/doc/docs/api-plugin/pagination.mdx (hunk 3 above)
Comment (wwwy3y3) [reply packages/doc/docs/api-plugin/pagination.mdx]: addressed in e265cf5eb3ab7f6230a5e4c419181f346638a6e4

File: packages/doc/docs/data-privacy/cls.mdx (hunk 2 above)
Comment (wwwy3y3) [reply packages/doc/docs/data-privacy/cls.mdx]: addressed in 023e474a6ca741d158c248b701be526af1a8f3de

File: packages/doc/docs/get-started/first-api.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (wwwy3y3) [reply packages/doc/docs/get-started/first-api.mdx:37]: addressed in 023e474a6ca741d158c248b701be526af1a8f3de

I simply mentioned "quick-start" in other places in this doc.

File: packages/doc/docs/api-plugin/rate-limit.mdx (hunk 1 above)
Comment (wwwy3y3) [reply packages/doc/docs/api-plugin/rate-limit.mdx]: addressed in abb1013b1527ec81a31b194761ec56ee6dc009c3

File: packages/doc/docs/api-plugin/format.mdx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+| Name     | Default | Type   | Description                                                                                          |
+| -------- | ------- | ------ | ---------------------------------------------------------------------------------------------------- |
+| default  | json    | string | The default format used when the request path doesn't specify a format type. Must be `json` or `csv`. |
+| formats  | -       | list   | A list of supported response format types. Include the format type if you need to use it in the path. |
+
+## Example Scenarios
+| Accept                                   | API URL Path        | options.default | options.formats  | response format type |
+| ---------------------------------------- | ----------------- | --------------- | ---------------- | -------------------- |
+| `q=0.9;application/json`                 | `/api/data`      | `json`          | `["json","csv"]` | json                 |
```
[the remark this answers is not in the document]
Comment (wwwy3y3) [reply packages/doc/docs/api-plugin/format.mdx]: addressed in abb1013b1527ec81a31b194761ec56ee6dc009c3

File: packages/doc/docs/connect/bigquery.mdx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+3. Create a new profile in `profiles.yaml` or in your profile files. For example:
+   :::info
+   You can choose one from `keyFilename` or `credentials` to use.
+
+   Your service account must have the following permissions to successfully execute queries.
+
+   - BigQuery Data Viewer
+   - BigQuery Job User
+
```
[the remark this answers is not in the document]
Comment (onlyjackfrost) [reply packages/doc/docs/connect/bigquery.mdx:28]: [Storage Object Admin](https://cloud.google.com/storage/docs/access-control/iam-roles)

**Review by fredalai — COMMENTED** [review]
1. The link is broken for the Catalog's Next page link (https://vulcan-sql-document-git-feature-refi-1fa767-vulcan-sql-document.vercel.app/docs/catalog/catalog-intro)
2. For the Catalog issue number, use #160 instead of #77.
3. ((Not sure will be in this PR to do it or not)) Update related catalog feature figures, and describe more detailed information. E.g., Adding descriptions for each API.

**Comment by wwwy3y3** [comment]
@fredalai 
> ((Not sure will be in this PR to do it or not)) Update related catalog feature figures, and describe more detailed information. E.g., Adding descriptions for each API.

Will not be done in this PR.

File: packages/doc/docs/connect/bigquery.mdx (hunk 8 above)
[the remark this answers is not in the document]
Comment (wwwy3y3) [reply packages/doc/docs/connect/bigquery.mdx:28]: Added

**Pull request merged**